src/Containers/Microsoft.NET.Build.Containers/Layer.cs

```diff
@@ -44,111 +45,78 @@ public static Layer FromDescriptor(Descriptor descriptor)
     }
 
     public static Layer FromDirectory(string directory, string containerPath, bool isWindowsLayer)
-    {
-        var fileList =
-            new DirectoryInfo(directory)
-            .EnumerateFiles("*", SearchOption.AllDirectories)
-            .Select(fsi =>
-                    {
-                        string destinationPath = Path.Join(containerPath, Path.GetRelativePath(directory, fsi.FullName)).Replace(Path.DirectorySeparatorChar, '/');
-                        return (fsi.FullName, destinationPath);
-                    });
-        return FromFiles(fileList, isWindowsLayer);
-    }
-
-    public static Layer FromFiles(IEnumerable<(string path, string containerPath)> fileList, bool isWindowsLayer)
     {
         long fileSize;
         Span<byte> hash = stackalloc byte[SHA256.HashSizeInBytes];
         Span<byte> uncompressedHash = stackalloc byte[SHA256.HashSizeInBytes];
 
-        // this factory helps us creating the Tar entries with the right attributes
-        PaxTarEntry CreateTarEntry(TarEntryType entryType, string containerPath)
+        // Docker treats a COPY instruction that copies to a path like `/app` by
```

Comment:
I moved the operations that operate on the `containerPath` together to the start of the method.